### ISSUE TITLE
Full path to file should be given in EXIF error message

### ISSUE THREAD

**Issue by snicolai-blog** [gohugoio/hugo#12693]
<pre>
$ hugo version
hugo v0.130.0+extended darwin/arm64 BuildDate=2024-07-29T13:51:56Z VendorInfo=brew
</pre>

I use page objects in my blog, and the new EXIF reader prints error message files relative to the page object.  I'd really like it to print them relative to the blog root, so I know which file.  For example:

```
WARN  images: "/fox.jpg": BitsPerSample: expected []interface {}, got uint16:
 ```

would be much better as:

```
WARN  images: "content/posts/part1/part2/fox.jpg": BitsPerSample: expected []interface {}, got uint16:
```

If it makes sense to strip the content/posts portion off to shorten the error message, that's fine.

**Comment by bep** [comment]
You are right, but that info isn't easily available where this is printed, which is why I got lazy. The important info for me (to fix this) here is the info to the right.

**Comment by nickjj** [comment]
It would be great if this could also apply to layout errors that are due to a content file being the culprit.

For example, I got this error recently:

```
Error: error building site: render: failed to render pages: render of "page" failed: "/app/layouts/blog/single.html:3:21": execute of template failed: template: blog/single.html:3:21: executing "main" at <.Fragments.HeadingsMap>: nil pointer evaluating *tableofcontents.Fragments.HeadingsMap
```

The problem is I have 500 posts and 300 drafts and there's no indication in the error which page is causing it. There's only a reference to a layout file to a function that is being used by those 800 posts.

Similarly I've gotten errors related to a `nil` reference from a missing image where the error only mentioned the `baseof` layout file being the issue but it's really due to a missing image in one of the 800 posts. It would be nice to see which content page that image is being referenced from.

**Comment by snicolai-blog** [comment]
Another option is to add "while processing <path to page>" to the error message.

**Comment by nickjj** [comment]
Replying to remove staleness since this is still an issue.

**Comment by cweiske** [comment]
>  If this is a feature request, and you feel that it is still relevant and valuable, please tell us why.

I'm upgrading to a newer hugo version and something broke. Now I know that
> render of "page" failed

but not which of my markdown files has a problem, which is really hindering debugging.

Source of the "render of page failed" error:
https://github.com/gohugoio/hugo/blob/13a95b9c0167c832fe2babf961ac5cad22e3a1e4/hugolib/site.go#L1735
